salt/client/ssh/ssh_py_shim.py

```diff
@@ -292,7 +292,32 @@ def main(argv):  # pylint: disable=W0613
             os.makedirs(OPTIONS.saltdir)
             cache_dir = os.path.join(OPTIONS.saltdir, "running_data", "var", "cache")
             os.makedirs(os.path.join(cache_dir, "salt"))
-            os.symlink("salt", os.path.relpath(os.path.join(cache_dir, "venv-salt-minion")))
+            os.symlink(
+                "salt", os.path.relpath(os.path.join(cache_dir, "venv-salt-minion"))
+            )
+        if os.path.exists(OPTIONS.saltdir) and (
+            "SUDO_UID" in os.environ or "SUDO_GID" in os.environ
+        ):
+            try:
+                sudo_uid = int(os.environ.get("SUDO_UID", -1))
+            except ValueError:
+                sudo_uid = -1
+            try:
+                sudo_gid = int(os.environ.get("SUDO_GID", -1))
+            except ValueError:
+                sudo_gid = -1
+            dstat = os.stat(OPTIONS.saltdir)
+            if (sudo_uid != -1 and dstat.st_uid != sudo_uid) or (
+                sudo_gid != -1 and dstat.st_gid != sudo_gid
+            ):
+                os.chown(OPTIONS.saltdir, sudo_uid, sudo_gid)
+                for dir_path, dir_names, file_names in os.walk(
+                    OPTIONS.saltdir, followlinks=True
```

Comment:
Is `followlinks=True` really required here?

Reply:
There is at least one symbolic link to the directory inside `OPTIONS.saltdir`, by default it's empty, and it should be safe to remove it, potentially if someone put improper link to outside of the saltdir to somewhere else it could be a problem, I'll check if it's safe to just remove it :+1:

Reply:
Fixed, thanks for pointing it :+1: , potentially it could be a security issue.